addons/website_sale/static/src/js/website_sale_utils.js

```diff
@@ -32,7 +35,68 @@ function animateClone($cart, $elem, offsetTop, offsetLeft) {
     }
 }
 
+/**
+ * Allows to propose the existing products from the database.
+ *
+ * @param {ServicesMixin|Widget} self - an element capable to trigger an RPC
+ * @param {jQuery} $input
+ */
+function autocompleteWithProducts(self, $input, data) {
+    $input.autocomplete({
+        source: function (request, response) {
+            return self._rpc({
+                route: '/website_sale/get_products_info',
+                params: {
+                    term: request.term,
+                    limit: 7,
+                    product_options: {
+                        'display_description': data.displayDescription,
+                        'display_price': data.displayPrice,
+                        'display_image': data.displayImage,
+                        'order_by': data.orderBy,
+                    }
+                },
+            }).then((function (resources) {
+                this.products_options = resources.products_options;
+                var orderBy = this.products_options.order_by;
+                if (orderBy === 'order_by_name_AZ') {
+                    resources.products_info.sort();
+                } else if (orderBy === 'order_by_name_ZA') {
+                    resources.products_info.reverse();
+                } else if (orderBy === 'order_by_price_low') {
+                    orderByPrice('low_to_high');
+                } else {
+                    orderByPrice('high_to_low');
+                }
+                response(resources.products_info);
+
+                function orderByPrice(order) {
+                    resources.products_info.sort(function (a, b) {
+                        if (order === 'low_to_high') {
+                            return a.price - b.price;
+                        } else {
+                            return b.price - a.price;
+                        }
+                    });
+                }
+            }).bind(this));
+        }
+    }).autocomplete('instance')._renderItem = function (ul, item) {
+        var descMaxLength = 60;
+        if (item.description.length > descMaxLength) {
+            item.description = item.description.substring(0, descMaxLength) + '...';
```

Comment:
Do you know if it is possible to do this via jinja filter in XML or something like that ? It could be better

Reply:
jinja is for mail and stuff. You should do this in SCSS, check the `o-text-overflow` mixin